akka-stream/src/main/scala/akka/stream/scaladsl/OperationAttributes.scala

```diff
@@ -57,6 +63,7 @@ object OperationAttributes {
   final case class InputBuffer(initial: Int, max: Int) extends Attribute
   final case class Dispatcher(dispatcher: String) extends Attribute
   final case class SupervisionStrategy(decider: Supervision.Decider) extends Attribute
+  final case class MaterializationName(n: String) extends Attribute
```

Comment:
Not that it is hacky. My main problem is that attributes are user api while this materialized name is specific to the materializer.